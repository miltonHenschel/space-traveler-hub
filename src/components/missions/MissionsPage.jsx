import { useSelector, useDispatch } from 'react-redux';
import { useEffect } from 'react';
// import { v4 as uuidv4 } from 'uuid';
import { fetchMissionsFromAPI } from '../../redux/missions/missionsSlice';

import styles from '../../styles/Missions.module.css';
import Missions from './Missions';

function MissionsPage() {
  const missions = useSelector((state) => state.missions);
  const isLoading = useSelector((state) => state.missions.isLoading);

  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchMissionsFromAPI());
  }, [dispatch]);

  return (
    <table className={styles.table}>
      <thead>
        <tr className={styles.theadTr}>
          <th className={styles.th}>Mission</th>
          <th className={styles.th}>Description</th>
          <th className={styles.th}>Status</th>
          <th>
            <div />
          </th>
        </tr>
      </thead>
      <tbody>
        {isLoading && <tr className={styles.tr}>Loading...</tr>}
        {!isLoading
          && missions.missionsItems.map((mission) => (
            <Missions
              key={mission.mission_id}
              mission_id={mission.mission_id}
              mission_name={mission.mission_name}
              description={mission.description}
              reserved={mission.reserved}
            />
          ))}
      </tbody>
    </table>
  );
}

export default MissionsPage;
