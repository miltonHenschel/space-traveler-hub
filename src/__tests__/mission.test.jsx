import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import MissionsPage from '../components/missions/MissionsPage';
import store from '../redux/Store';

it('Test Missions renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <MissionsPage />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
