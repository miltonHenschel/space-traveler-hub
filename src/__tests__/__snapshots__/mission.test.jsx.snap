// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Missions renders correctly 1`] = `
<table
  className="table"
>
  <thead>
    <tr
      className="theadTr"
    >
      <th
        className="th"
      >
        Mission
      </th>
      <th
        className="th"
      >
        Description
      </th>
      <th
        className="th"
      >
        Status
      </th>
      <th>
        <div />
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      className="tr"
    >
      Loading...
    </tr>
  </tbody>
</table>
`;
